### PR TITLE
TPA for self-signed certs support

### PR DESCRIPTION
**Pull request by carlosthe19916**
https://issues.redhat.com/browse/TC-1050

## Objective
- Make the Trustification Helm Chart to be instantiated in a CRC local Openshift installation. [CRC](https://developers.redhat.com/products/openshift-local/overview?source=sso) has self signed certificates, therefore a good way of reproducing the ticket's error description.

## Context
- The error described in the ticket implies that SSO/Keycloak is being used in the OCP cluster and WE ARE NOT USING AN EXTERNAL SSO PROVIDER

## Understanding the error
- spog-api, vexination-api, bombastic-api, and vy11-api; all are trying to get a connection to the SSO/Keycloak Openshift Route/Ingress
- The SSO/Keycloak Router/Ingress is something like `https://mydomain....` and since we are using CRC, then the `HTTPS` thing will fail as not trusted certificates are being used

## Solution

Current: pods are making connections to an insecure Route

```mermaid
graph TD;
    spog-api-pod-->sso-route-https;
    vexination-api-pod-->sso-route-https;
    bombastic-api-pod-->sso-route-https;
    vy11-api-pod-->sso-route-https;
   sso-route-https-->browser
```

Solution: pod make connections to the K8s Service rather than the Route. Any service within the same cluster should be reachable by any pod.

```mermaid
graph TD;
    spog-api-pod-->sso-route-service;
    vexination-api-pod-->sso-route-service;
    bombastic-api-pod-->sso-route-service;
    vy11-api-pod-->sso-route-service;
   sso-route-service-->sso-route-https;   
   sso-route-https-->browser
```

## Know "limitation"
- spog-api, vexination-api, bombastic-api have an environment variable SWAGGER_UI_OIDC_ISSUER_URL in place which MUST be an Ingress/Route URL as it is used in the browser, therefore can not use the K8s Service url.

Recommendation: Remove all the Swagger thing from the helm charts. Why are we having it in a PRODUCTION helm chart? Nice to have it in development but never saw anything similar in production ready apps.

## "--auth-configuration"
spog-api, vexination-api, bombastic-api, and vy11-api all have the `--auth-configuration` argument in the pod initialization. This argument OVERWRITES the ENV VARIABLES of SSO set up on the Deployment. I'm removing it but I'd like to know your opinions about it.

## Test in CRC
- Start CRC
- Prepare the project
```shell
oc new-project trustification
oc adm policy add-scc-to-user privileged -z infrastructure-postgresql &&
oc adm policy add-scc-to-user privileged -z infrastructure-minio &&
oc adm policy add-scc-to-user privileged -z infrastructure-keycloak &&
oc adm policy add-scc-to-user privileged -z infrastructure-kafka-provisioning &&
oc adm policy add-scc-to-user privileged -z infrastructure-kafka
```

- Prepare vars:
```shell
export NAMESPACE=trustification
APP_DOMAIN=-trustification.apps-crc.testing
```

- Move your terminal to the directory `deploy/k8s` and then:
- Install infraestructure:
```shell
helm upgrade --install -n $NAMESPACE \
            infrastructure charts/trustification-infrastructure \
            --wait --timeout 10m \
            --values values-minikube.yaml \
            --set-string keycloak.ingress.hostname=sso$APP_DOMAIN \
            --set-string appDomain=$APP_DOMAIN
```

- Install Trustification:
```shell
helm upgrade --install -n $NAMESPACE \
            trustification charts/trustification \
            --wait --timeout 10m \
            --values values-minikube.yaml \
            --set-string appDomain=$APP_DOMAIN \
            --set-string ssoServiceName=infrastructure-keycloak.trustification \
            --set modules.bombasticApi.disableSwaggerOidc=true \
            --set modules.spogApi.disableSwaggerOidc=true \
            --set modules.vexinationApi.disableSwaggerOidc=true
```

### CODE REVIEW

File: deploy/k8s/charts/trustification/values.schema.yaml [hunk 1]
```diff
@@ -18,6 +18,11 @@ properties:
       When using OpenShift, this should be the application domain, when can be evaluated by running the following
       command: `oc -n openshift-ingress-operator get ingresscontrollers.operator.openshift.io default -o jsonpath='{.status.domain}'`.
 
+  ssoServiceName:
```
Comment (ctron) [inline deploy/k8s/charts/trustification/values.schema.yaml:21]: I don't like the name and location of this variable: a) sso is not necessarly keycloak b) Not everyone uses "OCP" or knows that that means. c) I don't think it properly explains what should go into that variable.

File: deploy/k8s/charts/trustification/templates/services/bombastic/api/030-Deployment.yaml [hunk 2]
```diff
@@ -43,8 +43,6 @@ spec:
             - "--index-mode"
             - {{ $mod.module.indexMode | default "file" | quote }}
 
-            - "--auth-configuration"
```
Comment (ctron) [inline deploy/k8s/charts/trustification/templates/services/bombastic/api/030-Deployment.yaml:46]: Why is the full authentication configuration being removed? What about the other configuration options in there?

File: deploy/k8s/charts/trustification/templates/helpers/_oidc.tpl [hunk 3]
```diff
@@ -38,7 +38,11 @@ Arguments (dict):
   {{- include "trustification.common.envVarValue" $client.clientSecret | nindent 2 }}
 
 - name: OIDC_PROVIDER_ISSUER_URL
+  {{- if .root.Values.ssoServiceName }}
```
Comment (ctron) [inline deploy/k8s/charts/trustification/templates/helpers/_oidc.tpl:41]: Why isn't it just possible to use the `.oidc.issuerUrl` variable in the values file?

**Review by ctron — CHANGES_REQUESTED** [review]

**Comment by carlosthe19916** [comment]
Replaced by https://github.com/trustification/trustification/pull/1251

**Pull request closed without merging**